src/System.IO.FileSystem/src/System/IO/Win32FileStreamAsyncResult.cs

```diff
@@ -207,7 +207,8 @@ internal void CallUserCallback()
                     // Set completedSynchronously to false, since it's on another 
                     // thread, not the main thread.
                     _completedSynchronously = false;
-                    Task.Run(() => CallUserCallbackWorker());
+                    Task.Factory.StartNew(state => ((FileStreamAsyncResult)state).CallUserCallbackWorker(),
```

Comment:
thanks, though in general I haven't added a comment explaining this particular pattern, since one of the main purposes of object state is to pass state through to the delegate in order to avoid closure/delegate allocations.  The rest of the invocation is just an expansion of what Task.Run itself does; this Task.Run overload is just a wrapper that's basically this same call to StartNew, in a sense I'm just changing what overload is used.